parcels/grid.py

```diff
@@ -146,79 +176,113 @@ def lon_particle_to_target(self, lon):
-                if signdt == -1 and self.ti == 0 and (time - periods*(self.time_full[-1]-self.time_full[0])) == self.time[0] and f.time_periodic:
-                    self.ti = len(self.time)-1
+                if (
+                    signdt == -1
+                    and self.ti == 0
+                    and (time - periods * (self.time_full[-1] - self.time_full[0])) == self.time[0]
+                    and f.time_periodic
+                ):
+                    self.ti = len(self.time) - 1
                     periods -= 1
                 if signdt == -1 and self.ti > 0 and self.time_full[self.ti] == time:
                     self.ti -= 1
                 if self.ti >= len(self.time_full) - 1:
                     self.ti = len(self.time_full) - 2
 
-                self.time = self.time_full[self.ti:self.ti+2]
+                self.time = self.time_full[self.ti : self.ti + 2]
                 self.tdim = 2
                 if prev_time_indices is None or len(prev_time_indices) != 2 or len(prev_time_indices) != len(self.time):
-                    self.update_status = 'first_updated'
-                elif functools.reduce(lambda i, j: i and j, map(lambda m, k: m == k, self.time, prev_time_indices), True) and len(prev_time_indices) == len(self.time):
-                    self.update_status = 'not_updated'
-                elif functools.reduce(lambda i, j: i and j, map(lambda m, k: m == k, self.time[:1], prev_time_indices[:1]), True) and len(prev_time_indices) == len(self.time):
-                    self.update_status = 'updated'
+                    self.update_status = "first_updated"
+                elif functools.reduce(
+                    lambda i, j: i and j, map(lambda m, k: m == k, self.time, prev_time_indices), True
+                ) and len(prev_time_indices) == len(self.time):
+                    self.update_status = "not_updated"
+                elif functools.reduce(
+                    lambda i, j: i and j, map(lambda m, k: m == k, self.time[:1], prev_time_indices[:1]), True
+                ) and len(prev_time_indices) == len(self.time):
+                    self.update_status = "updated"
                 else:
-                    self.update_status = 'first_updated'
-            if signdt >= 0 and (self.ti < len(self.time_full)-2 or not f.allow_time_extrapolation):
-                nextTime_loc = self.time[1] + periods*(self.time_full[-1]-self.time_full[0])
+                    self.update_status = "first_updated"
+            if signdt >= 0 and (self.ti < len(self.time_full) - 2 or not f.allow_time_extrapolation):
+                nextTime_loc = self.time[1] + periods * (self.time_full[-1] - self.time_full[0])
             elif signdt < 0 and (self.ti > 0 or not f.allow_time_extrapolation):
-                nextTime_loc = self.time[0] + periods*(self.time_full[-1]-self.time_full[0])
+                nextTime_loc = self.time[0] + periods * (self.time_full[-1] - self.time_full[0])
         return nextTime_loc
```

Comment:
Can generally benefit from a refactor, but too large a task to do now

Reply:
Note that grid.py may be completely rewritten when we start work on unstructured grids anyways